Forge.TreeWalker/src/TreeWalkerSession.cs

```diff
@@ -321,18 +358,40 @@ public async Task<string> WalkTree(string treeNodeKey)
                     }
                     finally
                     {
-                        // Always call this callback, whether or not visit node was successful.
-                        await this.Parameters.Callbacks.AfterVisitNode(
+                        if (this.Parameters.CallbacksV2 != null)
+                        {
+                            // When CallbacksV2 is not null, continues with the new ITreeWalkerCallbacksV2.
```

Comment:
>// When CallbacksV2 is not null, continues with the new ITreeWalkerCallbacksV2. [](start = 28, length = 79)

Let's use the same comment in BeforeVisitNode and AfterVisitNode.

Reply:
"Use CallbacksV2 when available."

---
In reply to: [505912854](https://github.com/microsoft/Forge/pull/32#discussion_r505912854) [](ancestors = 505912854)

Reply:
fixed => // If applicable, use the new CallbacksV2 with ITreeWalkerCallbacksV2

---
In reply to: [505917661](https://github.com/microsoft/Forge/pull/32#discussion_r505917661) [](ancestors = 505917661,505912854)